### PR TITLE
SALTO-938 - SBQQ__CustomScript__c references

### PR DESCRIPTION
**Pull request by adamro**

### CODE REVIEW

File: packages/salesforce-adapter/src/transformers/reference_mapping.ts [hunk 1]
```diff
@@ -64,10 +66,14 @@ type MetadataTypeArgs = {
   type: string
   typeContext: ReferenceContextStrategyName
 }
+type MetadataParentArgs = {
+  parent?: string
```
Comment (netama) [inline packages/salesforce-adapter/src/transformers/reference_mapping.ts:70]: nit: I think `parent` doesn't need to be optional?

**Review by netama — APPROVED** [review]
:+1:

File: packages/salesforce-adapter/src/transformers/reference_mapping.ts (hunk 1 above)
Comment (adamro) [reply packages/salesforce-adapter/src/transformers/reference_mapping.ts:70]: There's a "default" of not having either `parent` nor `parentContext`, I kinda hacked it through using `PickOne` and making `parent` optional. It's either that or creating a `PickAtMostOne` I guess?

Comment (netama) [reply packages/salesforce-adapter/src/transformers/reference_mapping.ts:70]: Oh right :) 
Up to you, but might be a little clearer either with a `PickAtMostOne` implementation or with `type PickNone<T> = { [P in keyof T]?: never }` and 
```
& (PickOne<MetadataParentArgs, 'parent'> | PickOne<MetadataParentArgs, 'parentContext'> | PickNone<MetadataParentArgs>)
```
below?

Comment (netama) [reply packages/salesforce-adapter/src/transformers/reference_mapping.ts:70]: on second thought, I think the current solution should be good (maybe extending optional to `parentContext` too for symmetry) - so ignore the last comment :)

**Pull request merged**